tests/generation/test_generate.py

```diff
@@ -59,3 +59,93 @@ def test_model_generation_input_dimensions(neuron_model_path):
     # Using an incompatible input length
     with pytest.raises(ValueError, match="The input sequence length"):
         _test_model_generation(model, tokenizer, model.batch_size, input_length=model.max_length * 2)
+
+
+@is_inferentia_test
+@requires_neuronx
+def test_seq2seq_generation_beam(neuron_seq2seq_beam_path):
+    model = NeuronModelForSeq2SeqLM.from_pretrained(neuron_seq2seq_beam_path)
+    tokenizer = AutoTokenizer.from_pretrained(neuron_seq2seq_beam_path)
+    inputs = tokenizer("translate English to German: Lets eat good food.", return_tensors="pt")
+
+    # 1. max length
+    output = model.generate(**inputs, num_return_sequences=2, max_length=5)
+    assert len(output[0]) <= 5
+
+    # 2. min length
+    output = model.generate(**inputs, num_return_sequences=2, min_length=10)
+    assert len(output[0]) >= 10
+
+    # 3. max new tokens
+    output = model.generate(**inputs, num_return_sequences=2, max_new_tokens=5)
+    assert len(output[0].unique()) <= 5
+
+    return output
+
+
+@is_inferentia_test
+@requires_neuronx
+def test_seq2seq_generation_beam_with_optional_outputs(neuron_seq2seq_beam_path_with_optional_outputs):
+    model = NeuronModelForSeq2SeqLM.from_pretrained(neuron_seq2seq_beam_path_with_optional_outputs)
+    tokenizer = AutoTokenizer.from_pretrained(neuron_seq2seq_beam_path_with_optional_outputs)
+    inputs = tokenizer("translate English to German: Lets eat good food.", return_tensors="pt")
+
+    output = model.generate(
+        **inputs,
+        num_return_sequences=1,
+        max_length=20,
+        output_scores=True,
+        output_attentions=True,
+        output_hidden_states=True,
+        return_dict_in_generate=True,
+    )
+    assert "scores" in output
+    assert "decoder_attentions" in output
+    assert "cross_attentions" in output
+    assert "decoder_hidden_states" in output
+
+    return output
+
+
+@is_inferentia_test
+@requires_neuronx
+def test_seq2seq_generation_greedy(neuron_seq2seq_greedy_path):
+    model = NeuronModelForSeq2SeqLM.from_pretrained(neuron_seq2seq_greedy_path)
+    tokenizer = AutoTokenizer.from_pretrained(neuron_seq2seq_greedy_path)
+    inputs = tokenizer("translate English to German: Lets eat good food.", return_tensors="pt")
+
+    # 1. max length
+    output = model.generate(**inputs, num_return_sequences=1, max_length=5)
```

Comment:
Unless I am mistaken, generate returns the input tokens also.

You could adapt your max_new_token test accordingly to remove the length of inputs.input_ids, then check the number.

Reply:
at least for t5, input tokens is not part of generated sequence.

https://github.com/huggingface/transformers/blob/1dfa03f12b3748dc7e9c2b5ada40c3401ada23a5/src/transformers/generation_utils.py#L925-L929

Reply:
I modified the assertion, it did not count `decoder_start_token_id` in

https://github.com/huggingface/transformers/issues/17868